uefi-macros/src/lib.rs

```diff
@@ -211,3 +211,39 @@ pub fn entry(args: TokenStream, input: TokenStream) -> TokenStream {
     };
     result.into()
 }
+
+#[proc_macro]
+pub fn cstr8(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let input: LitStr = parse_macro_input!(input);
+    let input = input.value();
+    match input
+        .chars()
+        .map(u8::try_from)
+        .collect::<Result<Vec<u8>, _>>()
+    {
+        Ok(c) => {
+            quote!(unsafe { ::uefi::CStr8::from_bytes_with_nul_unchecked(&[ #(#c),* , 0 ]) }).into()
+        }
+        Err(_) => syn::Error::new_spanned(input, "Invalid character")
+            .into_compile_error()
+            .into(),
+    }
+}
+
+#[proc_macro]
+pub fn cstr16(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let input: LitStr = parse_macro_input!(input);
+    let input = input.value();
+    match input
+        .chars()
+        .map(|c| u8::try_from(c).map(|c| c as u16))
```

Comment:
I think we can allow a wider range of characters here like this:

```rust
.map(|c| u16::try_from(u32::from(c)).map(|c| c as u16))
```

Then you can add a test like this demonstrating a character outside the `u8` range:

```rust
assert_eq!(
    crate::prelude::cstr16!("Б").to_u16_slice_with_nul(),
    [1041, 0]
);
```